tests/lint/crds_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lint
```

Comment:
Great to get the rolling here 🥇 

Could we home this in `staticanalysis` or `conventions`? I fear `lint` is a bit too overused. And/Alternatively we can add a readme for kind of tools we intend to add for this package

Reply:
+1 though I think this is not blocking. WDYT @acpana ?

Reply:
Renamed to `apichecks`.  I figure we can add in our fuzzers / field coverage tests in here also.  We can always rename again if we come up with a better name!